data_safe_haven/infrastructure/programs/sre/workspaces.py

```diff
@@ -133,6 +137,17 @@ def __init__(
             storage_account_data_private_sensitive_name=props.storage_account_data_private_sensitive_name,
         ).apply(lambda kwargs: self.read_cloudinit(**kwargs))
 
+        # Deploy maintenance configuration
+        maintenance_configuration = SREMaintenanceComponent(
+            "sre_maintenance",
+            stack_name,
+            SREMaintenanceProps(
+                location=props.location,
+                workspaces_resource_group_name=resource_group.name,
```

Comment:
If the maintenance component is only applied to VMs under workspaces, perhaps it doesn't need its own resource group? You could just pass `resource_group_name=resource_group.name` here and deploy it into the same resource group as the workspace VMs.